api/rosetta/data_network_integration_test.go

```diff
@@ -0,0 +1,523 @@
+// Copyright 2021 Optakt Labs OÜ
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not
+// use this file except in compliance with the License. You may obtain a copy of
+// the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations under
+// the License.
+
+// +build integration
+
+package rosetta_test
+
+import (
+	"encoding/json"
+	"net/http"
+	"regexp"
+	"testing"
+
+	"github.com/labstack/echo/v4"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/optakt/flow-dps/api/rosetta"
+	"github.com/optakt/flow-dps/models/convert"
+	"github.com/optakt/flow-dps/models/dps"
+	"github.com/optakt/flow-dps/rosetta/configuration"
+	"github.com/optakt/flow-dps/rosetta/identifier"
+)
+
+func TestAPI_Status(t *testing.T) {
+
+	db := setupDB(t)
+	api := setupAPI(t, db)
+
+	const (
+		lastHeight    = 425
+		oldestBlockID = "d47b1bf7f37e192cf83d2bee3f6332b0d9b15c0aa7660d1e5322ea964667b333"
+	)
+
+	lastBlock := knownHeader(425)
+
+	request := rosetta.StatusRequest{
+		NetworkID: defaultNetwork(),
+	}
+
+	rec, ctx, err := setupRecorder(statusEndpoint, request)
+	require.NoError(t, err)
+
+	err = api.Status(ctx)
+	assert.NoError(t, err)
+
+	assert.Equal(t, http.StatusOK, rec.Result().StatusCode)
+
+	// unpack response
+	var status rosetta.StatusResponse
+	require.NoError(t, json.Unmarshal(rec.Body.Bytes(), &status))
+
+	// verify current block
+	assert.Equal(t, status.CurrentBlockID.Index, lastBlock.Height)
+	assert.Equal(t, status.CurrentBlockID.Hash, lastBlock.ID().String())
+	assert.Equal(t, status.CurrentBlockTimestamp, convert.RosettaTime(lastBlock.Timestamp))
+
+	// verify oldest block
+	assert.Equal(t, status.OldestBlockID.Hash, oldestBlockID)
+
+	// this is actually omitted from JSON, due to height being zero, and JSON having the 'omitempty' tag
+	assert.Equal(t, status.OldestBlockID.Index, uint64(0))
```

Comment:
I don't understand what this is referring to 🤔

Reply:
It's just noting the fact that here we're verifying that the index of the oldest block is 0.
Though technically, the height is 0, and JSON with the `omitempty` tag actually did not send anything in the response.

So it's more that we're confirming that we weren't sent anything than that we were sent a height of 0.